File: onlinepayments-sdk/src/main/java/com/onlinepayments/sdk/client/android/model/validation/ValidationRule.java
/*
 * Copyright 2020 Global Collect Services B.V
 */

package com.onlinepayments.sdk.client.android.model.validation;

import com.onlinepayments.sdk.client.android.model.PaymentRequest;

/**
 * Interface for ValidationRule.
 */
public interface ValidationRule {

	/**
	 * Validate method which validates a text.
	 *
	 * @param value the text to be validated
	 *
	 * @return true, if the text is valid; false, if the text is invalid
	 *
	 */
    boolean validate(String value);

	boolean validate(PaymentRequest paymentRequest, String fieldId);

}

File: sync/Worldline/repo/onlinepayments-sdk/src/main/kotlin/com/onlinepayments/sdk/client/android/model/validation/ValidationRule.kt
/*
 * Do not remove or alter the notices in this preamble.
 *
 * Copyright © 2025 Worldline and/or its affiliates.
 *
 * All rights reserved. License grant and user rights and obligations according to the applicable license agreement.
 *
 * Please contact Worldline for questions regarding license and user rights.
 */
package com.onlinepayments.sdk.client.android.model.validation

import com.onlinepayments.sdk.client.android.model.PaymentRequest

/**
 * Interface for ValidationRule.
 */
interface ValidationRule {
    /**
     * Validate method which validates a text.
     *
     * @param paymentRequest the fully filled [PaymentRequest] that will be used for doing a payment
     * @param fieldId the ID of the field to which to apply the current validator
     *
     * @return true, if the text is valid; false, if the text is invalid
     */
    fun validate(paymentRequest: PaymentRequest, fieldId: String): Boolean
}
